### ISSUE TITLE
Different (worse) results on validation split in the second run

### ISSUE THREAD

**Issue by mosharafhossain** [nyu-mll/jiant#1244]
I was just checking this example (training RoBERTa on mrpc.): https://github.com/nyu-mll/jiant/blob/master/examples/notebooks/jiant_Basic_Example.ipynb

When I train and evaluate the model I got the below results.
{
  "aggregated": 0.8837525685229688,
  "mrpc": {
    "loss": 0.5343004568265035,
    "metrics": {
      "major": 0.8837525685229688,
      "minor": {
        "acc": 0.8651960784313726,
        "f1": 0.9023090586145649,
        "acc_and_f1": 0.8837525685229688
      }
    }
  }
}

In my second run, I just made do_train=False like below. 
 
run_args = main_runscript.RunConfiguration(
    jiant_task_container_config_path="./run_configs/mrpc_run_config.json",
    output_dir="./runs/mrpc",
    model_type="roberta-base",
    model_path="./models/roberta-base/model/roberta-base.p",
    model_config_path="./models/roberta-base/model/roberta-base.json",
    model_tokenizer_path="./models/roberta-base/tokenizer",
    learning_rate=1e-5,
    eval_every_steps=500,
    do_train=False,
    do_val=True,
    do_save=True,
    force_overwrite=True,
)

I was hoping that the model will use the best fine-tuned model saved in ./runs/mrpc/best_model.p during the first run.  
However, I received the below results:
{
  "aggregated": 0.15808823529411764,
  "mrpc": {
    "loss": 0.7289315530887017,
    "metrics": {
      "major": 0.15808823529411764,
      "minor": {
        "acc": 0.3161764705882353,
        "f1": 0.0,
        "acc_and_f1": 0.15808823529411764
      }
    }
  }
}

Is it like I am missing something or this is a bug? Thank you.

**Comment by jeswan** [comment]
The second run should use the `model_path` of the trained model. If following the example, your second command should be amended as following:

```
run_args = main_runscript.RunConfiguration(
...
**model_path="./runs/mrpc/best_model.p",**
...
)
```
